lib/matplotlib/tests/test_widgets.py

```diff
@@ -254,6 +254,27 @@ def test_CheckButtons():
     check.disconnect(cid)
 
 
+def check_TextBox():
+    def submit(text):
+        tool.set_val('x**1')
+
+    def change(text):
+        tool.color = '1.0'
+    ax = get_ax()
+    tool = widgets.TextBox(ax, 'Evaluate', color='.95', initial='x**2')
+    assert tool.text == 'x**2'
+    tool.on_submit(submit)
+    tool.on_text_change(change)
+    tool.begin_typing(tool.text)
```

Comment:
It doesn't really make sense to enter text into the box, and not verify that that actually ended up in the box, but instead, _replace_ the text entirely on the submit handler.

If you want to verify that the handlers were called, then set a global, or use a `Mock` object that counts calls. And to verify that the setters work, just call them directly (and assert a check after.)

Reply:
Yes thank you so much, I have modified according to your suggestions.